vmm_core/virt_support_x86emu/src/emulate.rs

```diff
@@ -801,6 +805,47 @@ impl<T: EmulatorSupport, U> EmulatorCpu<'_, T, U> {
                 },
             })
     }
+
+    pub fn handle_guest_memory_error(
+        &mut self,
+        gva: u64,
+        gpa: u64,
+        err: &GuestMemoryError,
+        user_mode: bool,
+    ) -> bool {
+        match err.kind() {
+            guestmem::GuestMemoryErrorKind::VtlProtected => {
```

Comment:
You were probably right in a previous PR that one day we'll need to include the VTL in this error kind, or make separate cases for VTL1 and VTL2.